commands/ls.go

```diff
@@ -48,20 +51,125 @@ func runLsBug(cmd *cobra.Command, args []string) error {
 
 	allIds := backend.QueryBugs(q)
 
-	for _, id := range allIds {
+	bugExcerpt := make([]*cache.BugExcerpt, len(allIds))
+	for i, id := range allIds {
 		b, err := backend.ResolveBugExcerpt(id)
 		if err != nil {
 			return err
 		}
+		bugExcerpt[i] = b
+	}
+
+	switch lsOutputFormat {
+	case "plain":
+		return lsPlainFormatter(backend, bugExcerpt)
+	case "json":
+		return lsJsonFormatter(backend, bugExcerpt)
+	case "default":
+		return lsDefaultFormatter(backend, bugExcerpt)
+	default:
+		return fmt.Errorf("unknown format %s", lsOutputFormat)
+	}
+}
 
+type JSONBug struct {
+	Id           string    `json:"id"`
+	HumanId      string    `json:"human_id"`
+	CreationTime time.Time `json:"creation_time"`
+	LastEdited   time.Time `json:"last_edited"`
+
+	Status       string         `json:"status"`
+	Labels       []bug.Label    `json:"labels"`
+	Title        string         `json:"title"`
+	Actors       []JSONIdentity `json:"actors"`
+	Participants []JSONIdentity `json:"participants"`
+	Author       JSONIdentity   `json:"author"`
+
+	Comments int               `json:"comments"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+type JSONIdentity struct {
+	Id      string `json:"id"`
+	HumanId string `json:"human_id"`
+	Name    string `json:"name"`
+	Login   string `json:"login"`
+}
+
+func lsJsonFormatter(backend *cache.RepoCache, bugExcerpts []*cache.BugExcerpt) error {
+	for _, b := range bugExcerpts {
+		jsonBug := JSONBug{
+			b.Id.String(),
+			b.Id.Human(),
+			time.Unix(b.CreateUnixTime, 0),
+			time.Unix(b.EditUnixTime, 0),
+			b.Status.String(),
+			b.Labels,
+			b.Title,
+			[]JSONIdentity{},
+			[]JSONIdentity{},
+			JSONIdentity{},
+			b.LenComments,
+			b.CreateMetadata,
+		}
+
+		if b.AuthorId != "" {
+			author, err := backend.ResolveIdentityExcerpt(b.AuthorId)
+			if err != nil {
+				return err
+			}
+
+			jsonBug.Author.Name = author.DisplayName()
+			jsonBug.Author.Login = author.Login
+			jsonBug.Author.Id = author.Id.String()
+			jsonBug.Author.HumanId = author.Id.Human()
+		} else {
+			jsonBug.Author.Name = b.LegacyAuthor.DisplayName()
+			jsonBug.Author.Login = b.LegacyAuthor.Login
+		}
+
+		for _, element := range b.Actors {
+			actor, err := backend.ResolveIdentityExcerpt(element)
+			if err != nil {
+				return err
+			}
+
+			jsonBug.Actors = append(jsonBug.Actors, JSONIdentity{
+				actor.Id.String(),
+				actor.Id.Human(),
+				actor.Name,
+				actor.Login,
+			})
+		}
+
+		for _, element := range b.Participants {
+			participant, err := backend.ResolveIdentityExcerpt(element)
+			if err != nil {
+				return err
+			}
+			jsonBug.Participants = append(jsonBug.Participants, JSONIdentity{
+				participant.Id.String(),
+				participant.Id.Human(),
+				participant.DisplayName(),
+				participant.Login,
+			})
+		}
+
+		jsonObject, _ := json.MarshalIndent(jsonBug, "", "    ")
+		fmt.Printf("%s\n", jsonObject)
```

Comment:
printing bugs one by one means the output is not a well formed JSON document. It should be an array.